File: Begginers/Python/numero_perfeito_1164.py
entrada = int(input())

for j in range(entrada):
    N = int(input())
    lista = []

    for i in range(1, N):
        if N % i == 0:
           lista.append(i)
       
    soma = sum(lista)

    if soma == N:
       print(f"{N} eh perfeito")
    else:
        print(f"{N} nao eh perfeito")

File: Begginers/C++/numero_perfeito_1164.cpp
#include <iostream>

int main() {
    int entrada;
    
    scanf("%d", &entrada);

    for (int j = 0; j < entrada; j++) {
        int N;
        scanf("%d", &N);
        int soma = 0;

        for (int i = 1; i < N; i++) {
            if (N % i == 0) {
                soma += i;
            }
        }

        if (soma == N) {
            printf("%d eh perfeito\n", N);
        } else {
            printf("%d nao eh perfeito\n", N);
        }
    }

    return 0;
}
